introduction/java_output_formatting/Solution.java

```diff
@@ -0,0 +1,39 @@
+import java.util.Scanner;
+
+public class Solution {
+  static private final int REQUIRED_LENGTH = 15;
+  static private final int MAX_ONE_DIGIT_NUMBER = 9; 
+  static private final int MAX_TWO_DIGITS_NUMBER = 99;
+  static private final String ONE_DIGIT_NUMBER_PREFIX = "00";
+  static private final String TWO_DIGITS_NUMBER_PREFIX = "0";
+  
+  public static String formatOutput(String line, String numberPrefix, int number) {
+    String result = line;
+    for (int j = 0; j < REQUIRED_LENGTH - line.length(); j++) {
+      result += " ";
+    }
+    result += numberPrefix;
+    result += Integer.toString(number);
```

Comment:
`result += ..` shouldn't work. Then you have to use `result = result + ..`.

And in that case you don't need explicit conversion, you can just do 
`result = result + number;`

Also instead of several additions to a string you can use [`String.format`](https://dzone.com/articles/java-string-format-examples)